openpype/hosts/maya/api/pipeline.py

```diff
@@ -426,6 +468,49 @@ def on_before_save():
     return lib.validate_fps()
 
 
+def after_file_open():
+    """Check if there is a user opening the file"""
+    log.info("Running callback on checking the lock file...")
+
+    # add the lock file when opening the file
+    filepath = current_file()
+
+    if not is_workfile_locked(filepath):
+        create_workfile_lock(filepath)
+
+    else:
+        username = get_username(filepath)
+        reminder = cmds.window(title="Reminder", width=400, height=30)
+        cmds.columnLayout(adjustableColumn=True)
+        cmds.separator()
+        cmds.columnLayout(adjustableColumn=True)
+        comment = " %s is working the same workfile!" % username
+        cmds.text(comment, align='center')
+        cmds.text(vis=False)
+        cmds.rowColumnLayout(numberOfColumns=3,
+                             columnWidth=[(1, 300), (2, 100)],
+                             columnSpacing=[(2, 10)])
+        cmds.separator(vis=False)
+        quit_command = "cmds.quit(force=True);cmds.deleteUI('%s')" % reminder
+        cmds.button(label='Ok', command=quit_command)
+        cmds.showWindow(reminder)
+
+
+def on_before_close():
+    """Delete the lock file after user quitting the Maya Scene"""
+    log.info("Closing Maya...")
```

Comment:
Same as logging comment below - this can be more descriptive. Best if this explains what it is about to do right?

Also, I really wonder how reliable the "maya exiting callback" will be for this feature. Maya crashes a lot. We need to make sure that won't get in the way for users if maya force quits, breaks, crashes and you try to re-open the same scene.